spec/integrations/sidekiq_spec.rb

```diff
@@ -66,4 +66,90 @@ def perform(value)
       expect(error_handlers.size).to eq(1)
     end
   end
+
+  context "ensuring data reset" do
+    context "v2" do
+      before do
+        Sidekiq::Testing.inline!
+        stub_const('Sidekiq::VERSION', '2.0.0')
+        Sidekiq::Testing.server_middleware do |chain|
+          chain.add ::Bugsnag::Sidekiq
+        end
+      end
+
+      it "resets with an exception" do
+        expect(Bugsnag.configuration).to receive(:set_request_data).with(:sidekiq, hash_including(:queue => "default"))
+        expect(Bugsnag.configuration).to receive(:clear_request_data).at_least(:once)
+        begin
+          Worker.perform_async(-0)
+          fail("shouldn't be here")
```

Comment:
It seems more appropriate to have something like `expect { Worker.perform_async(-0) }.to raise_error` instead of this begin/rescue (similar comment for similar lines of code below)